ddtrace/internal/ci_visibility/coverage.py

```diff
@@ -0,0 +1,126 @@
+import contextlib
+from itertools import count
+from itertools import groupby
+import json
+import os
+from typing import Iterable
+from typing import List
+from typing import Tuple
+
+from ddtrace import config
+from ddtrace.internal import compat
+from ddtrace.internal.logger import get_logger
+
+from .constants import COVERAGE_TAG_NAME
+
+
+log = get_logger(__name__)
+
+try:
+    from coverage import Coverage
+    from coverage import version_info as coverage_version
+    from coverage.numbits import numbits_to_nums
+
+    # this public attribute became private after coverage==6.3
+    EXECUTE_ATTR = "_execute" if coverage_version > (6, 3) else "execute"
+except ImportError:
+    Coverage = None  # type: ignore[misc,assignment]
+    EXECUTE_ATTR = ""
+
+
+def enabled():
+    if config._ci_visibility_code_coverage_enabled:
+        if compat.PY2:
```

Comment:
does this mean that code coverage will be incompatible with python 2?

Reply:
Yes

Reply:
Done!